tests/playwright/Report.md

```diff
@@ -0,0 +1,267 @@
+
+- Configuring and Setting up of Wikimedia Core to run on CI, in this case on Github actions.
+- Implementation of all MediaWiki Core tests in Playwright
+- Running MediaWiki Core tests in WebdriverIO
+- Analysis of both WebdriverIO and Playwright tests in terms of speed and stability
+
+## WebdriverIO
+
+[WebdriverIO](https://github.com/webdriverio/webdriverio) is a test automation framework that allows you to run tests based on the Webdriver protocol and Appium automation technology. WebdriverIO is known to be;
+
+- Extendable: Adding helper functions, or more complicated sets and combinations of existing commands is simple and really useful
+- Compatible: WebdriverIO can be run on the WebDriver Protocol for true cross-browser testing as well as Chrome DevTools Protocol for Chromium-based automation using Puppeteer.
+- Feature Rich: The huge variety of community plugins allows you to easily integrate and extend your setup to fulfill your requirements.
+
+The above points are just a few of WebdriverIO capability. To explore more WebdriverIO capabilities, you can check out the [WebdriverIO](https://webdriver.io/) documentation and the [Wikipedia Repo](https://www.mediawiki.org/wiki/Selenium).
+
+WebdriverIO provides the ability and options to run commands in both asynchronous and synchronous operations. For asynchronous operations, you can use the JavaScript usual async/await, however for synchronous, it can be done through [node-fibers](https://www.npmjs.com/package/fibers). Both of these options have their own benefits and issues which can be found on the [WebdriverIO](https://webdriver.io/docs/sync-vs-async) website.
+
+### Sample WebdriverIO test code
+
+This code snippet navigates to wikipedia.org in Chrome and saves the screenshot.
+
+```js
+const { remote } = require('webdriverio');
+
+(async () => {
+ const browser = await remote({
+        capabilities: {
+            browserName: 'chrome'
+        }
+  })
+  await browser.url('https://www.wikipedia.org/')
+
+  await browser.getTitle()
+
+  await browser.saveScreenshot('./screenshot.png')
+  await browser.deleteSession();
+})();
+```
+
+### WebdriverIO and Mocha
+
+The WebdriverIO WDIO runner currently supports Mocha, Jasmine, and Cucumber frameworks and WikiMedia Core test is currently using Mocha framework. [Mocha](https://mochajs.org/) is a feature-rich JavaScript test framework running on Node.js and in the browser, making asynchronous testing simple and fun.
+
+WebdriverIO provides an adapter for Mocha which can easily be added as `@wdio/mocha-framework`.
+
+#### Sample WebdriverIO and Mocha code
+
+This code snippet navigates to wikipedia.org in Chrome and saves the screenshot.
+
+```js
+describe('Wikipedia home page', () => {
+    it('should display correct page title', () => {
+        browser.url('https://www.wikipedia.org/')
+        expect(browser).toHaveTitle('The Free Encyclopedia')
+    })
+});
+```
```

Comment:
This sample code doesn't have any `require` statements. It also doesn't save the screenshot.